compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -118,7 +118,19 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
         case Nil  => body1
         case tags => tpd.Block(tags, body1).withSpan(body.span)
 
-    super.transformQuotation(body2, quote)
+    quote match {
+      case Apply(fn1 @ TypeApply(fn0, targs), _) =>
+        val heal = new TypeMap {
+          def apply(tp: Type) = tp match {
+            case AnnotatedType(parent, annot) => AnnotatedType(mapOver(parent), annot)
+            case _ => healTypeOfTerm(fn1.srcPos)(tp)
+          }
+        }
```

Comment:
I'm wondering why this is not already handled in the recursive call? Can it be handled in `transform`? I'm wondering if it's possible to avoid the `heal` TypeMap here and move the logic to `healTypeOfTerm` -- as there's also a TypeMap there.

Reply:
Moved into `healTypeOfTerm`